compute/client_library/snippets/tests/test_compute_reservation.py

```diff
@@ -46,6 +56,63 @@ def cleanup():
     return reservation
 
 
+@pytest.fixture(scope="session")
+def vm_instance(request):
+    """the fixture should create a VM instance"""
+    boot_disk = compute_v1.AttachedDisk()
+    boot_disk.auto_delete = True
+    boot_disk.boot = True
+    boot_disk.initialize_params = compute_v1.AttachedDiskInitializeParams(
+        source_image="projects/debian-cloud/global/images/family/debian-11"
+    )
+
+    additional_disk = compute_v1.AttachedDisk()
+    additional_disk.auto_delete = True
+    additional_disk.boot = False
+    additional_disk.initialize_params = compute_v1.AttachedDiskInitializeParams(
+        disk_size_gb=375
+    )
+    additional_disk.interface = "SCSI"
+
+    instance = create_instance(
+        project_id=PROJECT_ID,
+        zone=ZONE,
+        instance_name=INSTANCE_NAME,
+        disks=[boot_disk, additional_disk],
+        machine_type=MACHINE_TYPE,
+    )
+
+    def cleanup():
```

Comment:
Nit: we could use `yield` style fixtures for cleanup. The code looks cleaner and is easier to read.

```py
@pytest.fixture(scope="session")
def vm_instance():
    instance = create_instance(...)
    yield instance
    try:
        delete_instance(...)
    except ...
```

See https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#:~:text=Adding%20finalizers%20directly,requires%20a%20bit%20more%20verbosity.

Reply:
Thanks for the suggestion! 🙂 I’ve made the changes and switched to yield-style fixtures for cleanup👌